lib/packwerk/unnecessary_dependency/validator.rb

```diff
@@ -0,0 +1,47 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Packwerk
+  module UnnecessaryDependency
+    class Validator
+      extend T::Sig
+      include Packwerk::Validator
+
+      Result = Packwerk::Validator::Result
+
+      sig { override.params(package_set: PackageSet, configuration: Configuration).returns(Result) }
+      def call(package_set, configuration)
+        results = T.let([], T::Array[Result])
+        return Result.new(ok: true) if !YAML.load_file('packwerk.yml')['prevent_unnecessary_dependencies']
```

Comment:
There should probably be a check that this file was generated when `bin/packwerk check` was run with no arguments, i.e. over the whole codebase.

Otherwise, this will produce invalid results.

Alternatively, it can work on a pack-by-pack basis.